lib/http_ntlm.c

```diff
@@ -141,6 +141,7 @@ CURLcode Curl_output_ntlm(struct Curl_easy *data, bool proxy)
   const char *passwdp;
   const char *service = NULL;
   const char *hostname = NULL;
+  const char *localhostname;
```

Comment:
This needs to be initialized to NULL to avoid uninitialized variable use in the `proxy` case.